File: software/examples/rust/example_input.rs
use std::{error::Error, io};

use tinkerforge::{io16_bricklet::*, ip_connection::IpConnection};

const HOST: &str = "localhost";
const PORT: u16 = 4223;
const UID: &str = "XYZ"; // Change XYZ to the UID of your IO-16 Bricklet.

fn main() -> Result<(), Box<dyn Error>> {
    let ipcon = IpConnection::new(); // Create IP connection.
    let io = Io16Bricklet::new(UID, &ipcon); // Create device object.

    ipcon.connect((HOST, PORT)).recv()??; // Connect to brickd.
                                          // Don't use device before ipcon is connected.

    // Get current value from port A as bitmask.
    let value_mask_a = io.get_port('a').recv()?;
    println!("Value Mask (Port A): {:b}", value_mask_a);

    // Get current value from port B as bitmask.
    let value_mask_b = io.get_port('b').recv()?;
    println!("Value Mask (Port B): {:b}", value_mask_b);

    println!("Press enter to exit.");
    let mut _input = String::new();
    io::stdin().read_line(&mut _input)?;
    ipcon.disconnect();
    Ok(())
}

File: software/examples/javascript/ExampleInput.js
var Tinkerforge = require('tinkerforge');

var HOST = 'localhost';
var PORT = 4223;
var UID = 'XYZ'; // Change XYZ to the UID of your IO-16 Bricklet

var ipcon = new Tinkerforge.IPConnection(); // Create IP connection
var io = new Tinkerforge.BrickletIO16(UID, ipcon); // Create device object

ipcon.connect(HOST, PORT,
    function (error) {
        console.log('Error: ' + error);
    }
); // Connect to brickd
// Don't use device before ipcon is connected

ipcon.on(Tinkerforge.IPConnection.CALLBACK_CONNECTED,
    function (connectReason) {
        // Get current value from port A as bitmask
        io.getPort('a',
            function (valueMaskA) {
                console.log('Value Mask (Port A): ' + valueMaskA.toString(2));
            },
            function (error) {
                console.log('Error: ' + error);
            }
        );

        // Get current value from port B as bitmask
        io.getPort('b',
            function (valueMaskB) {
                console.log('Value Mask (Port B): ' + valueMaskB.toString(2));
            },
            function (error) {
                console.log('Error: ' + error);
            }
        );
    }
);

console.log('Press key to exit');
process.stdin.on('data',
    function (data) {
        ipcon.disconnect();
        process.exit(0);
    }
);
